src/main/java/org/broadinstitute/hellbender/tools/reference/CheckReferenceCompatibility.java

```diff
@@ -0,0 +1,284 @@
+        TableColumnCollection columns = new TableColumnCollection(Arrays.asList("Reference", "Compatibility"));
+        try(CheckReferenceCompatibilityTableWriter writer = output == null
+                ? new CheckReferenceCompatibilityTableWriter(new OutputStreamWriter(System.out), columns)
+                : new CheckReferenceCompatibilityTableWriter(output.toPath(), columns)
+        ){
+            writer.writeComment(String.format("Current Reference: %s", dictionaryName));
+            writer.writeHeaderIfApplies();
+            for(CompatibilityRecord record : referenceCompatibilities){
+                writer.writeRecord(record);
+            }
+        }
+        catch(IOException exception){
+            throw new UserException.CouldNotCreateOutputFile(output, "Failed to write output table.", exception);
+        }
+    }
+
+    private boolean dictionaryHasMD5s(SAMSequenceDictionary dictionary){
+        for (SAMSequenceRecord record : dictionary.getSequences()) {
+            String md5FromDict = record.getMd5();
+            if (md5FromDict == null) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /** Evaluate the compatibility of a given pair of references when MD5s are present.
+     *
+     * @param refPair ReferencePair containing the dictionary file and a reference it is being compared against
+     * @param table
+     */
+    private void evaluateCompatibility(ReferencePair refPair, ReferenceSequenceTable table){
+        EnumSet<ReferencePair.Status> status = refPair.getAnalysis();
+        if(status.contains(ReferencePair.Status.EXACT_MATCH)){
+            referenceCompatibilities.add(new CompatibilityRecord(refPair.getRef2(), "Compatible, the sequence dictionaries exactly match"));
+        }
+        else if(status.contains(ReferencePair.Status.SUBSET) && status.size() == 1){
+            referenceCompatibilities.add(new CompatibilityRecord(refPair.getRef2(),String.format("Compatible, the sequence dictionary in %s is a subset of the %s reference sequence dictionary. Missing sequence(s): %s", refPair.getRef1AsString(), refPair.getRef2AsString(), getMissingSequencesIfSubset(dictionaries.get(refPair.getRef2())))));
+        }
+        else{
+            referenceCompatibilities.add(new CompatibilityRecord(refPair.getRef2(), String.format("Not compatible. Status: %s. Run CompareReferences tool for more information on reference differences.", status)));
+        }
+    }
+
+    /** Evaluate the compatibility of a given pair of references when MD5s are not present.
+     *
+     * Note: compatibility calls are made based on sequence name and length, alone, and are therefore imprecise.
+     *
+     * @param referenceDict SAMSequenceDictionary to be compared against the key sequence file
+     * @param referenceDictPath path to provided dictionary
+     */
+    private void evaluateCompatibility(SAMSequenceDictionary referenceDict, GATKPath referenceDictPath){
```

Comment:
i would not overload this method name when these are processing two different datatypes. I would name them evaluateCompatabilityWithMD5Table for example.